libdino/src/application.vala

```diff
@@ -58,6 +58,9 @@ public interface Application : GLib.Application {
         });
         shutdown.connect(() => {
             stream_interactor.connection_manager.make_offline_all();
+#if _WIN32
+            Dino.Util.CoUninitialize();
```

Comment:
According to documentation:
> A thread must call CoUninitialize once for each successful call it has made to the CoInitialize or CoInitializeEx function, including any call that returns S_FALSE.

This is only once per application start.

Reply:
Uhm.. Here it says that every call to `CoInitialize` must be balanced with a call to `CoUninitialize`:
https://docs.microsoft.com/pt-br/windows/win32/api/objbase/nf-objbase-coinitialize

Those COM stuff is confusing. I changed the code to make it call `CoUninitialize` as soon as possible after calling `CoInitialize`, not sure if this is correct...
Maybe I guess I could just remove this `CoInitialize` code, not sure if there is much use case to open something with components that require COM (such as shell extensions).

---

How many threads there is in Dino? Is the thread that starts/shutdowns Dino the same thread that calls ShellExecute?

Reply:
Dino has multiple threads, but most of it happens in a single main thread. Start and shutdown should be in the same thread. But I guess your change is sufficient, even though I am not an expert on Windows things.